detection_rules/rule_validators.py

```diff
@@ -192,11 +192,17 @@ def validate_integration(
                 integration_schema_data["integration"],
             )
             integration_schema = integration_schema_data["schema"]
+            stack_version = integration_schema_data["stack_version"]
 
             # Add non-ecs-schema fields
             for index_name in data.index:
                 integration_schema.update(**ecs.flatten(ecs.get_index_schema(index_name)))
 
+            # Add custom schema fields for appropriate stack version
+            if data.index:
+                for index_name in data.index:
+                    integration_schema.update(**ecs.flatten(ecs.get_custom_index_schema(index_name, stack_version)))
```

Comment:
Should we gate this behind the CUSTOM_RULES_DIR env variable?

Reply:
Yes! Good idea! Will add :+1:

Reply:
Added